### PR TITLE
spotdl: init at 3.5.0

### PR DESCRIPTION
**Pull request by dotlambda**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117782` at e50867bd run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built successfully:</summary>
  <ul>
    <li>python38Packages.pytube</li>
    <li>python38Packages.rapidfuzz</li>
    <li>python39Packages.pytube</li>
    <li>python39Packages.rapidfuzz</li>
    <li>spotdl</li>
  </ul>
</details>

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117782` at e50867bd run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built successfully:</summary>
  <ul>
    <li>python38Packages.pytube</li>
    <li>python38Packages.rapidfuzz</li>
    <li>python39Packages.pytube</li>
    <li>python39Packages.rapidfuzz</li>
    <li>spotdl</li>
  </ul>
</details>

File: pkgs/development/python-modules/pytube/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  src = fetchFromGitHub {
+    owner = "pytube";
+    repo = "pytube";
+    rev = "v${version}";
+    sha256 = "0wjnhzmv1lxs3yb46ywmbqj3j60srsrs02f7c5v4d2zaw1mnc8zh";
+  };
+
+  propagatedBuildInputs = [
+    typing-extensions
+  ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pytube/default.nix:24]:
```suggestion
```

see https://github.com/pytube/pytube/blob/v10.6.0/Pipfile#L6

Comment (dotlambda) [reply pkgs/development/python-modules/pytube/default.nix:24]: Weird. I was looking at https://github.com/pytube/pytube/blob/v10.6.0/setup.py#L29.

Comment (dotlambda) [reply pkgs/development/python-modules/pytube/default.nix:24]: https://github.com/pytube/pytube/pull/965

**Pull request merged**